engine/src/mblandroiddc.cpp

```diff
@@ -2817,6 +2817,30 @@ void MCAndroidDisableOpenGLMode(void)
 
 ////////////////////////////////////////////////////////////////////////////////
 
+static bool s_in_permission_dialog = false;
+static bool s_permission_granted = false;
+bool MCAndroidCheckRuntimePermission(MCStringRef p_permission)
+{
+    bool t_result;
+    s_in_permission_dialog = true;
+    MCAndroidEngineRemoteCall("askPermission", "bx", &t_result, p_permission);
+    
+    while (s_in_permission_dialog)
+        MCscreen -> wait(60.0, True, True);
```

Comment:
@livecodepanos I think it might be simpler to make this a blocking wait `MCscreen -> wait(60.0, False, True);` as if we are using a dispatching wait it's feasible that there could be another request during it and the implementation would need to be much more complicated to handle that.